### PR TITLE
[fix] add exception for the platform in network policy

### PR DESCRIPTION
**Pull request by facchettos**
**What issue type does this pull request address?** (keep at least one, remove the others) 
/kind bugfix

**What does this pull request do? Which issues does it resolve?** (use `resolves #<issue_number>` if possible) 
resolves eng-4121

**Please provide a short message that should be published in the vcluster release notes**
Fixed an issue where vcluster would not configure network policies correctly when trying to connect to the platform 

**What else do we need to know?**

### CODE REVIEW

File: config/config.go [hunk 1]
```diff
@@ -1438,7 +1438,8 @@ type LimitRange struct {
 
 type NetworkPolicy struct {
 	// Enabled defines if the network policy should be deployed by vCluster.
-	Enabled bool `json:"enabled,omitempty"`
+	Enabled         bool `json:"enabled,omitempty"`
+	PlatformEnabled bool `json:"platformEnabled,omitempty"`
```
Comment (FabianKramm) [inline config/config.go:1442]: Platform enabled is a little bit misleading, can we move this into `outgoingConnections.platform: true`?

File: chart/templates/networkpolicy.yaml [hunk 2]
```diff
@@ -95,6 +95,12 @@ spec:
           podSelector:
             matchLabels:
               k8s-app: kube-dns
+        {{- if .Values.policies.networkPolicy.platformEnabled }}
```
Comment (johannesfrey) [inline chart/templates/networkpolicy.yaml:98]: We want to toggle this separately and this shouldn't be just hard-coded like we did with dns when `.Values.policies.networkPolicy.enabled` is set to true?!

File: config/config.go (hunk 1 above)
Comment (facchettos) [reply config/config.go:1442]: I followed what was in the ticket

File: config/config.go [hunk 3]
```diff
@@ -1451,6 +1451,8 @@ type OutgoingConnections struct {
 	// to the pods matched by a NetworkPolicySpec's podSelector. The except entry describes CIDRs
 	// that should not be included within this rule.
 	IPBlock IPBlock `json:"ipBlock,omitempty"`
+
+	Platform bool
```
Comment (FabianKramm) [inline config/config.go:1455]: Please add a comment as well above the field

**Review by FabianKramm — APPROVED** [review]

**Pull request merged**